compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -858,6 +869,54 @@ public void visit(BLangSimpleVariable varNode) {
         transferForkFlag(varNode);
     }
 
+    private void analyzeTypeNode(BLangType typeNode, SymbolEnv env) {
+        if (typeNode.getKind() == NodeKind.USER_DEFINED_TYPE) {
+            return;
+        }
+        if (typeNode.getKind() == NodeKind.RECORD_TYPE && !((BLangRecordTypeNode) typeNode).analyzed) {
+            analyzeDef(typeNode, env);
+        }
+        switch (typeNode.type.tag) {
```

Comment:
Don't we have to check error, future, function, and typedesc too?

I assume objects will work because of the module-level type definitions that are added, but let's add tests for them too, if we don't have already.